node/src/ledger/ledger_effects.rs

```diff
@@ -219,13 +233,27 @@ fn next_read_requests_init<S: redux::Service>(store: &mut Store<S>) {
         }) else {
             continue;
         };
-        if store.dispatch(LedgerReadAction::Init { request }) {
-            if !is_streaming {
-                store.dispatch(P2pChannelsRpcAction::ResponsePending { peer_id, id });
-            } else {
-                store.dispatch(P2pChannelsStreamingRpcAction::ResponsePending { peer_id, id });
-            }
+
+        store.dispatch(LedgerReadAction::Init {
+            request,
+            callback: LedgerReadInitCallback::P2pChannelsResponsePending
+         {      callback: redux::callback!(on_ledger_read_init_p2p_channels_response_pending((is_streaming: bool, id: P2pRpcId, peer_id: PeerId)) -> crate::Action{
+                    if is_streaming {
+                        P2pAction::from(P2pChannelsStreamingRpcAction::ResponsePending {
+                            peer_id,
+                            id,
+                        })
+                                    } else {
```

Comment:
@0xMimir indentation is weird here

Reply:
Fixed, I am not sure why fmt formatted it that way.

Reply:
It doesn't handle code inside macros very well.